### PR TITLE
Add the lesson find use case to the DG

### PR DESCRIPTION
**Pull request by ethantyh692**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -505,6 +505,27 @@ Priorities: MVP (must have), 2 (nice to have), 3 (unlikely to have)
       Steps 1a1 to 1a2 are repeated until the data entered are correct.  
       Use case resumes from Step 2.
 
+**Use Case: UC08 - Find lessons with a specific name keyword**
+
+**MSS**:
+
+1. Tutor keys in a keyword in the required field to find lessons whose names contain this keyword.
```
Comment (Ma-Xizi) [inline docs/DeveloperGuide.md:512]: Find lessons with student names that contain this keyword?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -505,6 +505,27 @@ Priorities: MVP (must have), 2 (nice to have), 3 (unlikely to have)
...
+1. TutorEase lists the lessons with the given keyword.  
+   Use case ends.
+
+**Extensions**:
+
+* **1a**. TutorEase detects bad or wrongly formatted inputs.
```
Comment (Ma-Xizi) [inline docs/DeveloperGuide.md:518]: Should we give more details about different 'bad or wrongly formatted inputs' for different commands? etc. number , special characters?

Comment (ethantyh692) [reply docs/DeveloperGuide.md:518]: I thought that use case do not include a lot of implementation details? Let's see what the rest think

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ethantyh692) [reply docs/DeveloperGuide.md]: Fixed

**Review by Chronoxy — APPROVED** [review]
LGTM! 👍

**Pull request merged**